src/System.Device.Gpio/Interop/Unix/libgpiod/Interop.libgpiod.cs

```diff
@@ -96,30 +97,30 @@ internal partial class Interop
     /// <param name="value">New value.</param>
     /// <returns>0 if the operation succeeds. In case of an error this routine returns -1 and sets the last error number.</returns>
```

Comment:
For crypto interop I've been simplifying my callers (making them look a bit more C# than C):

```C#
[DllImport(LibgpiodLibrary)]
private static extern int gpiod_line_set_value(SafeLineHandle line, int value);

internal static void SetGpiodLineValue(SafeLineHandle line, int value)
{
    int ret = gpiod_line_set_value(line, value);

    if (ret < 0)
    {
        throw whatever();
    }
}
```

(If the fail isn't throw then make it bool, and normalize in the Interop layer).

That's a thing to consider here.  You could maintain the Shim-Style naming, and wrap when it makes sense, and set the Entrypoint property on the DllImport when you want to invoke direct.

e.g.

```C#
[DllImport(LibgpiodLibrary, SetLastError = true, Entrypoint = "gpiod_chip_get_line")]
internal static extern SafeLineHandle GetChipLineByOffset(SafeChipHandle chip, int offset);
```